### ISSUE TITLE
http+unix URL support for backends

### ISSUE THREAD

**Issue by sloonz** [containous/traefik#4881]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

Nginx allows reverse proxying to HTTP servers that do not listen on a TCP socket, but an Unix socket (`upstream backend { server unix:/tmp/backend.sock; }`).

Listening on Unix sockets is extremely useful in non-containerized environments (like the file provider), since it doesn’t require port allocation management, they honor filesystem ACLs, and they don’t risk raw port exposure on the internet in case of firewall/listening interface misconfiguration (in architectures when access control is the responsibility of the reverse proxy). 

This has already been mentionned #1396 but the issue has been closed due to a misunderstanding : @dtomcej and @ldez thought it was just an Unix variant of raw TCP reverse proxying (#10), but it is not : it is still HTTP reverse proxying, just on a different transport.

For reference, there is an `http.RoundTripper` implementation for doing HTTP roundtrip on Unix transport : https://github.com/tv42/httpunix/

**Comment by dtomcej** [comment]
@sloonz You are correct! When we closed #1396, most of the feature requests were centred around implementing layer 4 functionality. Now that Traefik v2 supports L4 and L7, this feature becomes even more interesting and powerful. Thank you for pointing this out!

**Comment by hostmit** [comment]
I vote up this feature! I've been waiting for it quite a while.

**Comment by fsdrw08** [comment]
Any update of this request?

**Comment by fsdrw08** [comment]
I would like to expose podman unix socket by traefik, by default, podman socket only serve as a unix socket.  
I understand that podman socket is able to serve as a background service to listen system tcp connection, but I just want to keep podman as lean as possible.

**Comment by shoenig** [comment]
Another use case - the imminent Nomad 1.5 release contains a [Task API](https://github.com/hashicorp/nomad/blob/v1.5.0-rc.1/website/content/api-docs/task-api.mdx#task-api) feature - whereby any task configured with a new `identity` block is given a unique workload idenitty and a Unix domain socket with which to communicate with the Nomad API.

We had begun looking into replacing this [NGINX reverse proxy](https://developer.hashicorp.com/nomad/tutorials/manage-clusters/reverse-proxy-ui) guide with one based on Traefik leveraging the new feature, but this issue is a blocker.

**Comment by rtribotte** [comment]
Hello @shoenig,

We understand that you want to expose Nomad's WebUI with Traefik, using this new Task API.
We took a look at the Task API documentation, but I can't find any mention of the WebUI that would also be exposed by this new API, can you share with us exactly the part of the documentation about it? Thanks!

**Comment by yegle** [comment]
Another use case is so that I can use Tailscale's forward auth server that was written for nginx: https://tailscale.com/blog/tailscale-auth-nginx/

The binary generates a unix domain socket, which works with nginx (obviously) and Caddy (https://caddyserver.com/docs/caddyfile/directives/forward_auth#tailscale).

**Comment by courtland** [comment]
> Another use case is so that I can use Tailscale's forward auth server that was written for nginx: https://tailscale.com/blog/tailscale-auth-nginx/
> 
> The binary generates a unix domain socket, which works with nginx (obviously) and Caddy (https://caddyserver.com/docs/caddyfile/directives/forward_auth#tailscale).

FWIW I came here for this. It's a shame there's no unix socket support, particularly given the new integration with Tailscale certs. ForwardAuth against Tailscale would be a great use case.

**Comment by unknowndevQwQ** [comment]
Another use case: I want to expose caddy, nginx, etc. to several socket files using traefik, because controlling the permissions of the socket files determines which processes have access to the corresponding services, as well as the use of systemd's socket unit, and I want to bind as few ports as possible.

By the way, unix socket files are more like the network layer where IPv4 or IPv6 resides. If it's not too much work to support it, I'd recommend making traefik fully support [unix domain socket files](https://serverfault.com/questions/124517/what-is-the-difference-between-unix-sockets-and-tcp-ip-sockets) everywhere.

**Comment by bendavis78** [comment]
I feel like the use case of "not having to use port numbers that could potentially conflict w/ other services" is a fairly solid reason to support this. Nginix has been able to do this for ages.
